shell-completion/bash/systemd-creds

```diff
@@ -0,0 +1,145 @@
+        { while read -r a b c; do echo " $a"; done }
+}
+
+_systemd_creds() {
+    local verb comps mode
+    local cur=${COMP_WORDS[COMP_CWORD]} prev=${COMP_WORDS[COMP_CWORD-1]} words cword
+    local -A OPTS=(
+        [STANDALONE]='-h --help --version --no-pager --no-legend
+                      -H -T --system --user -p --pretty --allow-null'
+        [ARG]='--tpm2-device
+               --tpm2-device-key
+               --tpm2-pcrs
+               --tpm2-public-key
+               --tpm2-public-key-pcrs
+               --tpm2-signature
+               --json
+               --transcode
+               --newline
+               --name
+               --timestamp
+               --not-after
+               --with-key
+               --uid'
+    )
+
+    local -A VERBS=(
+        [STANDALONE]='list setup'
+        [CAT]='cat'
+        [ENCRYPT]='encrypt'
+        [DECRYPT]='decrypt'
+    )
+
+    _init_completion || return
+
+    if __contains_word "--user" ${COMP_WORDS[*]}; then
+        mode=--user
+    elif __contains_word "--system" ${COMP_WORDS[*]}; then
+        mode=--system
+    else
+        mode=''
+    fi
+
+    if __contains_word "$prev" ${OPTS[ARG]}; then
+        case $prev in
+            --tpm2-public-key|--tpm2-signature)
+                comps=$(compgen -A file -- "$cur")
+                compopt -o filenames
+                ;;
+            --tpm2-device)
+                comps="auto list $(__get_tpm2_devices)"
+                ;;
+            --json)
+                comps=$( systemd-creds --json=help 2>/dev/null )
+                ;;
+            --transcode)
+                comps='base64 unbase64 hex unhex'
+                ;;
+            --with-key)
+                comps='host tpm2 host+tpm2 null auto auto-initrd'
+                ;;
+        esac
+        COMPREPLY=( $(compgen -W '$comps' -- "$cur") )
+        return 0
+    fi
+
+    if [[ -z ${verb-} && "$cur" = -* ]]; then
+        COMPREPLY=( $(compgen -W '${OPTS[*]}' -- "$cur") )
+        return 0
+    fi
+
+    for ((i=0; i < COMP_CWORD; i++)); do
+        if __contains_word "${COMP_WORDS[i]}" ${VERBS[*]} &&
+                ! __contains_word "${COMP_WORDS[i-1]}" ${OPTS[ARG]}; then
+            verb=${COMP_WORDS[i]}
+            break
+        fi
+    done
+
+    if [[ -z ${verb-} ]]; then
+        comps=${VERBS[*]}
+    elif __contains_word "$verb" ${VERBS[STANDALONE]}; then
+        comps=''
+    elif __contains_word "$verb" ${VERBS[CAT]}; then
+        if [[ $cur = -* ]]; then
+            comps='--help --version --no-pager --no-legend --user --system'
+        else
+            comps=$( __get_creds $mode )
+        fi
+    elif __contains_word "$verb" ${VERBS[ENCRYPT]}; then
+        if [[ $cur = -* ]]; then
+            comps='--help --version --no-pager --no-legend --uid --user --system --with-key -H -T --tpm2-device --tpm2-pcrs --tpm2-public-key --tpm2-public-key-pcrs --tpm2-signature --not-after --name --timestamp --pretty'
```

Comment:
Hm... This will be sooner or later outdated.
Also, `--user` or `--system` should not be specified when mode= is non-empty.

At least, please move the list to the beginning of the function, e.g.
```
local -A OPTS_FOR_VERB(
        [ENCRYPT] = ...
        [DECRYPT] = ...
)
```